my-app/src/main/java/com/JavaWebServer/app/ServerSocket.java

```diff
@@ -0,0 +1,41 @@
+package com.JavaWebServer.app;
+
+import java.io.IOException;
+
+public class ServerSocket implements InterfaceServerSocket {
+  private java.net.ServerSocket serverSocket;
+
+  public ServerSocket(int port) {
+    try{
+      this.serverSocket = new java.net.ServerSocket(port);
+    } catch(IOException e) {
+      throw new RuntimeException("can not open sever socket"+ e);
+    }
+  }
+
+  public ServerSocket (java.net.ServerSocket socket) {
+     
+    this.serverSocket = socket;
+
+  }
+
+ public ClientSocket accept() {
+    try {
+      return new ClientSocket(serverSocket.accept());
+    } catch(IOException e) {
+      throw new RuntimeException("can not accept Client Connnection"+ e);
+    }
+  }
+
+  public Boolean isClosed() {
+    return serverSocket.isClosed();
```

Comment:
If you are going to refer to your instance variables without `this` (which is fine) you should be consistent about it.